### ISSUE TITLE
Ability to specify a termination hook for Ray actors

### ISSUE THREAD

**Issue by arunmk** [ray-project/ray#10198]
### Describe your feature request
We need to do some cleanup steps when the refcounted, non-detached actors complete. These can be things like copying logs to some location, closing open handles etc. There is no clean way of doing this apart from implementing our own ref-counting mechanism.

/CC @stephanie-wang

**Comment by stephanie-wang** [comment]
@arunmk, would you be interested in contributing this? It should be relatively easy to add!

**Comment by arunmk** [comment]
@stephanie-wang sure, let me take a look.

**Comment by ericl** [comment]
You can already use `atexit`, no need for a custom hook here. See this example: https://github.com/ray-project/ray/blob/5dc4b6686e5885df7d53c6162773ece336994feb/rllib/examples/env/env_with_subprocess.py

**Comment by arunmk** [comment]
@ericl the thought was to call the destructor. I can use the atexit to call it definitely. Should I close this feature request?

**Comment by richardliaw** [comment]
It would be great to add an example to the ray documentation! I'm sure a lot of other folks would benefit.

**Comment by arunmk** [comment]
@ericl @richardliaw the thought was to allow the `__del__()` of the class to be called if available. I spent some amount of time debugging refcounts and leaks until I realized that the destructor would not be called.

There is no need of a *new* custom hook. I use 4 actor classes and I do see the need of calling this in all of the 4 classes. I can use the `atexit` approach but it seems like boilerplate code to my scenario. The `atexit` approach would also need careful design when dealing with with nested classes and nested actors.

Have there been many asks for this sort of feature? If so I think it is better to add the call of the destructor. If it is a very uncommon scenario, we can add documentation.
